src/target.c

```diff
@@ -108,61 +111,107 @@ stumpless_add_entry( struct stumpless_target *target,
   }
 
   clear_error(  );
+ 
+  if( stumpless_get_option( target, STUMPLESS_OPTION_PERROR ) == STUMPLESS_OPTION_PERROR ){
+	// setting is_log_formatted true concludes that STUMPLESS_OPTION_PERROR option is
+	// set, thus only checking this flag is enough for further checks
+	// NOTE : Calling the format_entry twice because if the buffer is for target who
+	// require unformatted entry would need not to call format_entry.
+	is_log_formatted = true;
+  	builder = format_entry( entry, target );
+  	if( !builder ) {
+    		return -1;
+  	}
+  	buffer = strbuilder_get_buffer( builder, &builder_length );
```

Comment:
You can simply do the write to the error stream directly from here, instead of in the logging call for each type of target. This will significantly reduce the amount of duplicated code, and make it easier for you to get full test coverage.